### PR TITLE
Add Helm Chart testing scripts

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Changes proposed in this pull request:

- Add Helm Chart testing scripts
- Fix LABEL entry on our Dockerfiles
- Add license boilerplate for our `clients` pkg
- Remove `values.schema.json` file from chart
- Fix helm chart installation (hardcoded URL for etcd cluster)
- Add missing entries in the chart to be compliant with official chart-testing validators
- Remove duplicated `platform` directory (content moved under /internal/platform pkg)
- Remove `contrib` dir, and leave only `hack` folder, reasons:
	Contrib meaning:
  > It is for software that has been contributed to the project, but which might not actually be maintained by the core developers. Naming it "contrib" or "Contrib" is a long-established convention

- remove the `heath` package from `pkg` directory, why:
  - In `pkg` we should put library code that's ok to use by external applications (e.g., /pkg/mypubliclib). So it's good to have there our `apis` and `clients` because it's our contract and we are supporting those libs and it's normal that someone will vendor that in own project. In case of `health` pkg we shouldn't expose such a contract because is not helm-broker domain and w do not want to support that. What's more, the `health` pkg will be removed when this will be merge: https://github.com/kubernetes-sigs/controller-runtime/pull/419. **We should think twice before putting something here :-)**

**Example output:**
```terminal
=== RUN   TestHelmBrokerChartHappyPath
--- PASS: TestHelmBrokerChartHappyPath (13.88s)
    helm_broker_test.go:94: ClusterAddonsConfigurations "test-hb-chart-zlr2p" is created
    helm_broker_test.go:121: ClusterAddonsConfigurations "test-hb-chart-zlr2p" has Ready state
    helm_broker_test.go:144: Helm Broker exposes Service [id: "a54abe18-0a84-22e9-ab34-d663bbce3d88"] from ClusterAddonsConfiguration test-hb-chart-zlr2p properly
    helm_broker_test.go:100: ClusterAddonsConfigurations "test-hb-chart-zlr2p" is deleted
PASS
ok  	github.com/kyma-project/helm-broker/test/charts	13.898s
```

## Continous Integration

I've executed that script manually on our CI, check results here: https://status.build.kyma-project.io/view/gcs/kyma-prow-logs/logs/post-master-hb-test/1175553385537998848

ProwJob definition was added in this PR: https://github.com/kyma-project/test-infra/pull/1501

Additionally, I've created an issue for test-infra https://github.com/kyma-project/test-infra/issues/1499 to support generic runner script. Currently, we have DIND option copied into our script. Check created issue for more info.

## Found issue

- [HelmBroker freezes and stops processing other (Cluster)AddonsConfigurations when ectd is down](https://github.com/kyma-project/helm-broker/issues/34)
- [Helm Broker README.md lacking a few things which are good to know when it comes to using/installing Helm Broker](https://github.com/kyma-project/helm-broker/issues/35)
- [Add generic runner script](https://github.com/kyma-project/test-infra/issues/1499)
--- 

**Related issue(s)**

- https://github.com/kyma-project/helm-broker/issues/23

### CODE REVIEW

File: test/charts/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Charts Tests / Smoke?
```
Comment (mszostok) [inline test/charts/README.md]:
```suggestion
# Charts Tests
```

File: test/charts/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+A chart directory contains tests that validate that our charts work as expected when they are installed. These tests also help the chart consumer understand what your chart is supposed to do.
+
+A test in a helm chart lives under the `templates/tests` directory and is a pod definition that specifies a container with a given command to run. The container should exit successfully (exit 0) for a test to be considered a success. The pod definition must contain the helm `helm.sh/hook: test-success` hook annotations.
```
Comment (klaudiagrz) [inline test/charts/README.md]:
```suggestion
Find the definition of a Pod that runs tests in the `templates/tests` directory. The Pod definition contains the Helm `helm.sh/hook: test-success` hook annotation, which means that the Pod is executed only when you run the `helm test {relase_name}` command.
```

File: test/charts/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+You can run the pre-defined tests in Helm on a release using the command `helm test <RELEASE_NAME>`. For a chart consumer, this is a great way to sanity check that their release of a chart (or application) works as expected.
```
Comment (klaudiagrz) [inline test/charts/README.md]:
```suggestion
```

File: before-commit.sh [hunk 4]
```diff
@@ -44,6 +44,15 @@ for binary in "${binaries[@]}"; do
 	fi
 done
 
+echo -e "${GREEN} Building charts test${NC}"
+${buildEnv} go test -v -c -o hb_chart_test ./test/charts/helm_broker_test.go
+goBuildResult=$?
+if [[ ${goBuildResult} != 0 ]]; then
+    echo -e "${RED}✗ go test -c ${binary} ${NC}\n$goBuildResult${NC}"
```
Comment (jasiu001) [inline before-commit.sh]: `binary` variable comes from the previous loop (`for binary in "${binaries[@]}"; do (...) done`) so the result of `echo` is `go build targz` instead e.g. `go build hb_chart_test` (or just `go build` if we run this part separately)

File: test/charts/helm_broker_test.go [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+			}
+		}
+		return false
+	}
+
+	assert.True(s.t, containService(s.cfg.ExpectedAddonID))
+	s.t.Logf("Helm Broker exposes Service [id: %q] from ClusterAddonsConfiguration %s properly", s.cfg.ExpectedAddonID, s.sampleClusterAddonsCfgName)
+}
+
+func waitForFuncAtMost(t *testing.T, fn func() error, timeoutDur time.Duration) {
```
Comment (jasiu001) [inline test/charts/helm_broker_test.go]: We use `k8s.io/apimachinery/pkg/util/wait` library in an integration test, is it any reason why you use a custom function here? You don't like apimachinery function ;) ?

**Pull request merged**